# Fix memory leak in test authdata server
#
# greet_kdc_sign() must always free if_relevant.
#
# patch comes from upstream
#	https://github.com/krb5/krb5/pull/732
#
# The fix should become available with next major release
#

--- a/src/plugins/authdata/greet_server/greet_auth.c
+++ b/src/plugins/authdata/greet_server/greet_auth.c
@@ -83,10 +83,9 @@ greet_kdc_sign(krb5_context context,
     if (code == 0) {
         krb5_free_authdata(context, enc_tkt_reply->authorization_data);
         enc_tkt_reply->authorization_data = tkt_authdata;
-    } else {
-        krb5_free_authdata(context, if_relevant);
     }
 
+    krb5_free_authdata(context, if_relevant);
     krb5_free_authdata(context, kdc_issued);
 
     return code;
